File: assg2/q3.py
# Reading two matrices and printing their products
with open('m.txt', 'r') as f:
    m = [[int(num) for num in line.split()] for line in f]

with open('n.txt', 'r') as f:
    n = [[int(num) for num in line.split()] for line in f]

a= [[1, 1, 2], [1, 1, 1], [1, 1, 3]]
p1=[[0,0,0], [0,0,0], [0,0,0]]
p2=[[0,0,0], [0,0,0], [0,0,0]]
for x in range(3):
    for i in range(3):
        for y in range(3):
            p1[x][i] = p1[x][i] + (m[x][y] * a[y][i])
print("Product of M*A: ",p1)
for x in range(3):
    for i in range(3):
        for y in range(3):
            p2[x][i] = p1[x][i] + (m[x][y] * n[y][i]) 
print("Product of M*N: ", p2)
#Product of M*A:  [[26, 26, 31], [15, 15, 31], [15, 15, 32]]
#Product of M*N:  [[37, 28, 39], [81, 27, 79], [48, 21, 56]]

File: assg2/q3.c
/* To read two matrices from text file and find the products. */
#include <stdio.h>
#include <stdlib.h>
int main()
{
    int matA[3] = {1,2,3};
    int i = 0;
    printf("Matrix A is: \n");
    for(i = 0; i<3; i++){
        printf("%d \n", matA[i]);
    }
    
    int matM[3][3] = {{0,0,0},{0,0,0},{0,0,0}};
    int matN[3][3] = {{0,0,0},{0,0,0},{0,0,0}};
    int matC[3][3] = {{0,0,0},{0,0,0},{0,0,0}};
    int matD[3] = {0,0,0};
    
    FILE *openM;
    
    openM = fopen("m.txt", "r+");
    for(int i = 0; i<3; i++){
        for(int j = 0; j<3; j++){
            fscanf(openM, "%d", &matM[i][j]  );
            
        }
        
    }
    fclose(openM);
    
    FILE *openN;
    
    openN = fopen("n.txt", "r+");
    for(int i = 0; i<3; i++){
        for(int j = 0; j<3; j++){
            fscanf(openN, "%d", &matN[i][j]  );
            
        }
        
    }
    fclose(openN);
    
    for (int i = 0; i < 3; i++) {
      for (int j = 0; j < 3; j++) {
        for (int k = 0; k < 3; k++) {
            matC[i][j] += matM[i][k]*matN[k][j];
        }
      }
    }
    printf("Matrix MN is: \n");
    for(int i =0; i<3; i++){
        for(int j = 0; j<3; j++){
        printf("%d ", matC[i][j]);
        
        }
    printf("\n");
    }
    
    for(int i = 0; i < 3; i++){
        for(int j =0; j<3; j++){
            matD[i] = matD[i] + matM[i][j]*matA[j];
        }
    }
        
    printf("Matrix MA is: \n");
    for(int i = 0; i<3; i++){
        printf("%d \n", matD[i]);
    }
    
    
    return 0;
}


/*Output
Matrix A is:                                                                                                                            
1                                                                                                                                       
2                                                                                                                                       
3                                                                                                                                       
Matrix MN is:                                                                                                                           
64 338 231                                                                                                                              
88 95 102                                                                                                                               
68 43 45                                                                                                                                
Matrix MA is:                                                                                                                           
50                                                                                                                                      
32                                                                                                                                      
22 
*/
